src/marshmallow_annotations/ext/dataclass.py

```diff
@@ -75,3 +82,11 @@ class Meta:
     @post_load
     def make_object(self, data):
         return self.opts.target(**data)
+
+
+def initvar_typehint(typehint: type):
+    return {_DATACLASS_METADATA_KEY: {"initvar-typehint": typehint}}
+
+
+def validator(validation_fn: Callable):
+    return
```

Comment:
I hope to make validations work like

```
@dataclasses.dataclass
class SomeClass:
    # required field with validator
    a: int = dataclasses.field(metadata=validator(lambda x: x > 50))
    # required load only field with validator
    b: dataclasses.InitVar[int] = dataclasses.field(metadata={
            **initvar_typehint(int),
            **validator(lambda x: x > 0),
})
    # required load only field with validator and other random metadata for non marshmallow-annotaitons purposes
    b: dataclasses.InitVar[int] = dataclasses.field(metadata={
            **{"Foo": "Bar'},
            **initvar_typehint(int),
            **validator(lambda x: x > 0),
})
```

But ran out of time tonight after getting `InitVar` rough draft done, and wanted feedback.

I can either pull the validation stuff out, or if you give me a few pointers I may be able to work it into this PR (I didn't see inline validations in attrib extension)

Reply:
Don't recall _why_ I didn't include validators in the attrs integration, but it definitely looks like an oversight on my part.

I'd say since dataclass doesn't include validation by default it should probably be held off here.